In [3]:
print('Takumi House & Kailen Santos')

Takumi House & Kailen Santos


In [4]:
from matplotlib import pyplot as plt
import matplotlib as mp
import numpy as np
import pandas as pd
from sklearn.cluster import KMeans
import sklearn.manifold
from scipy import cluster
import seaborn as sns
import sklearn.datasets as sk_data
import sklearn.metrics as metrics
import requests
import json
import nltk.classify.util
from nltk.classify import NaiveBayesClassifier
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
from nltk import word_tokenize
import re
import json
import pickle
%matplotlib inline

In [5]:
##load in pickled dataframe of all the reviews

#try:
#    with open('df_reviews','rb') as f:
#        df_reviews = pickle.load(f)
        
#except FileNotFoundError:
#    with open('df_reviews','wb') as f:
#        pickle.dump(df_reviews, f)

KeyboardInterrupt: 

In [6]:
##load in pickled stemmed data for the review text
try:
    with open('stemmed_data','rb') as f:
        stemmed_data = pickle.load(f)
        
except FileNotFoundError:
    stemmed_data = [" ".join(SnowballStemmer("english", ignore_stopwords=True).stem(word)  
        for sent in sent_tokenize(message)
        for word in word_tokenize(sent))
        for message in df_reviews['text']]
    
    with open('stemmed_data','wb') as f:
        pickle.dump(stemmed_data, f)

In [7]:
df = pd.read_csv('Restaurant_Inspections.csv',header=0)
# df = df[df.City == 'Las Vegas']
# df = (df[df['Inspection Result'] == 'Closed with Fees']).append(df[df['Inspection Result'] == 'Closed without Fees'])

df.shape

#with open ('Resaurant_Inspections.csv') as file:
#   for line in file:
#      if (line[6]=='Las Vegas' && line[]

C:\Users\Kailen\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2717: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


(123919, 23)

In [8]:
df = df[df.City == 'Las Vegas']
df.shape

(94700, 23)

In [9]:
categories = df['Category Name'].unique().tolist()
print(categories)

['Pantry', 'Special Kitchen', 'Garde Manger', 'Kitchen Bakery', 'Bar / Tavern', 'Buffet', 'Snack Bar', 'Restaurant', 'Banquet Support', 'Portable Unit', 'Banquet Kitchen', 'Portable Bar', 'Meat/Poultry/Seafood', 'Vegetable Prep', 'Caterer', 'Barbeque', 'Confection', 'Concessions', 'Produce Market', 'Institutional Food Service', 'Elementary School Kitchen', 'Self-Service Food Truck', 'Bakery Sales', 'Food Trucks / Mobile Vendor', 'Grocery Store Sampling', 'Beer Bar', 'Childcare Kitchens', "Farmer's Market", 'Main Kitchen', 'Frozen Meat Sales']


In [10]:
categories.remove('Elementary School Kitchen')
categories.remove('Grocery Store Sampling')
categories.remove('Farmer\'s Market')
categories.remove('Caterer')
categories.remove('Concessions')
categories.remove('Institutional Food Service')
categories.remove('Banquet Support')

In [11]:
df = df[df['Category Name'].isin(categories)]
df.shape

(91735, 23)

In [12]:
coord_strings = df['Location 1'].tolist()

In [13]:
import ast
xcoords = []
ycoords = []
for coord in coord_strings:
    xcoords.append(ast.literal_eval(coord)[0])
    ycoords.append(ast.literal_eval(coord)[1])

In [14]:
count = 0
for idx, ycoord in enumerate(ycoords):
    if ycoord > 0:
        ycoords[idx] = ycoord * -1
        count += 1
print('{} coordinates fixed.'.format(count))

574 coordinates fixed.


In [15]:
df['Coordinates'] = list(zip(xcoords, ycoords))
df.head()

Serial Number Permit Number                 Restaurant Name  \
11     DA0001715     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
12     DA0001716     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
13     DA0001728     PR0000427  CAESARS DIAMOND LOUNGE KITCHEN   
14     DA0001773     PR0000434         CAESARS BAKERY WAREWASH   
15     DA0001776     PR0000434         CAESARS BAKERY WAREWASH   

                    Location Name    Category Name                Address  \
11  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
12  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
13  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
14  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
15  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   

         City   State         Zip  Current Demerits  \
11  Las Vegas  Nevada  89109-8924                 3   
12  Las Vegas  Nevada  89109-8924                 3   
13  Las Vegas  Nevada  89109-8924                 8   
14  Las Vegas  Nevada  89109-8924                 3   
15  Las Vegas  Nevada  89109-8924                 3   

               ...             Employee ID     Inspection Type  \
11             ...               EE7000417  Routine Inspection   
12             ...               EE7000567  Routine Inspection   
13             ...               EE7000417  Routine Inspection   
14             ...               EE7000417  Routine Inspection   
15             ...               EE7000392  Routine Inspection   

   Inspection Demerits Inspection Grade Permit Status Inspection Result  \
11                   0                A           NaN         Compliant   
12                   2                A           NaN         Compliant   
13                   4                A           NaN         Compliant   
14                   7                A           NaN         Compliant   
15                   2                A           NaN         Compliant   

    Violations          Record Updated                  Location 1  \
11         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
12         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
13          14  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
14    14232831  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   
15         NaN  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   

                   Coordinates  
11  (36.1161559, -115.1750576)  
12  (36.1161559, -115.1750576)  
13  (36.1161559, -115.1750576)  
14  (36.1161559, -115.1750576)  
15  (36.1161559, -115.1750576)  

[5 rows x 24 columns]

In [16]:
coordinates_list = df['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])
df['X'] = xcoords
df['Y'] = ycoords

In [17]:
df = df[df['Y'] < -114.5]
df = df[df['Y'] > -115.6]
df_clean = df.reset_index(drop = True)

In [18]:
coordinates_list = df_clean['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])

In [19]:
df = df_clean[df_clean['Date Current']==df_clean['Inspection Date']]
df

Serial Number Permit Number                            Restaurant Name  \
63        DA0003413     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
196       DA0009343     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
254       DA0011310     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
317       DA0017310     PR0000631                  Golden Gate Beverage Cart   
348       DA001XLIF     PR0022088                         Marketon #1 - Deli   
436       DA0023482     PR0001171                         Aramark Nacho Cart   
503       DA0025038     PR0001066             Kentucky Fried Chicken #527621   
505       DA0025516     PR0001062         Kentucky Fried Chicken / Pizza Hut   
531       DA0027055     PR0001064            Kentucky Fried Chicken #X527030   
533       DA0027285     PR0001068             Kentucky Fried Chicken #527019   
557       DA0029175     PR0001074             Kentucky Fried Chicken #527024   
558       DA0029576     PR0001659                     K Mart #4369 Snack Bar   
592       DA0039728     PR0001891                CIRCUS MEXITALIA XPRESS BAR   
800       DA0055718     PR0000251                           Java in the Park   
891       DA0059162     PR0000322                Circle K Store #1248 - Deli   
1100      DA0093418     PR0002881                       VONS #1963 STARBUCKS   
1103      DA0093785     PR0002411                    Marie Callender's's #72   
1104      DA0093832     PR0002412             Marie Callender's's #72 Saloon   
1105      DA0093835     PR0002413      Marie Callender's's #72 Sunday Brunch   
1219      DA00E4NAR     PR0014994          OCEAN ONE BAR & GRILLE RESTAURANT   
1220      DA00KFO00     PR0104466                       WET N WILD SNACK BAR   
1221      DA00VEPXC     PR0006003                    HARD ROCK SANS POOL BAR   
1288      DA0104852     PR0001994                       White Cross Drugs #2   
1437      DA0113862     PR0002678                                Dairy Queen   
1493      DA0120049     PR0002749                  Carl's Jr Restaurant #883   
1551      DA0123453     PR0002800                      L a Express Ice Cream   
1743      DA0134595     PR0003488                          Burger King #2701   
1842      DA0138293     PR0003292                     7-Eleven #26637 - Deli   
1867      DA0142409     PR0003124                            Rascals Kitchen   
2014      DA0185680     PR0004240                                     Champs   
...             ...           ...                                        ...   
89526     DARL3H49O     PR0121917                       ANTOJITOS D.F. Y MAS   
89594     DAUDJGYGL     PR0119548                     Dotty's #138 SNACK BAR   
89650     DAVCPCHBE     PR0121950   HARD ROCK MAGIC MIKE'S LIVE BASEMENT BAR   
89662     DAVQQPOKX     PR0106985                 OFF THE STRIP PREP STORAGE   
89664     DAVSYBOOX     PR0120107      SUGAR FACTORY - UPSTAIRS DAIQUIRI BAR   
89667     DAVUOAFLR     PR0119884                            JACK IN THE BOX   
89669     DAVVC4M8B     PR0122607                WARES CANDY ICE CREAM TRUCK   
89685     DAWIIAJ0G     PR0122379     PLAZA HOTEL BRIGHTSIDE CAFE - COOKLINE   
89707     DAX7GAFWI     PR0121685                               INSPIRE PUSF   
89723     DAVE6DSQV     PR0121651                              FOOD TO HOMES   
89942     DAORFUG7L     PR0116193                   NUTRITION RUSH SNACK BAR   
90052     DAN46ZW1B     PR0122301                LEMON HEAVEN @ LVPO #2 PUSF   
90229     DALQAH6FS     PR0119843                              DISTILL - BAR   
90295     DAPSS2J5L     PR0120208          LA BONITA SUPERMARKETS RESTAURANT   
90389     DAA9MDFFM     PR0121622                               TAIL AND FIN   
90628     DA1763489     PR0116428                  MODERNA TEA @ FORUM SHOPS   
90672     DANI6SDQJ     PR0121470                        SUBWAY STORE #53283   
90686     DAXSEOVM5     PR0120738                          JABURRITOS @ LINQ   
90690 

In [20]:
df_clean = df.reset_index(drop=True)

In [21]:
df_agg = df_clean.groupby('Restaurant Name',as_index=False).agg(lambda x: x.tolist())

In [22]:
dfcategories = df_agg.columns
dfcategories

Index(['Restaurant Name', 'Serial Number', 'Permit Number', 'Location Name',
       'Category Name', 'Address', 'City', 'State', 'Zip', 'Current Demerits',
       'Current Grade', 'Date Current', 'Inspection Date', 'Inspection Time',
       'Employee ID', 'Inspection Type', 'Inspection Demerits',
       'Inspection Grade', 'Permit Status', 'Inspection Result', 'Violations',
       'Record Updated', 'Location 1', 'Coordinates', 'X', 'Y'],
      dtype='object')

In [23]:
# drop unnecessary columns for categorization step later
df_agg = df_agg.drop(['Serial Number'], axis = 1)
df_agg = df_agg.drop(['City'], axis = 1)
df_agg = df_agg.drop(['State'], axis = 1)
df_agg = df_agg.drop(['Zip'], axis = 1)
df_agg = df_agg.drop(['Inspection Time'], axis = 1)
df_agg = df_agg.drop(['Employee ID'], axis = 1)
df_agg = df_agg.drop(['Inspection Type'], axis = 1)
df_agg = df_agg.drop(['Location 1'], axis = 1)
df_agg = df_agg.drop(['Coordinates'], axis = 1)
df_agg = df_agg.drop(['Record Updated'], axis = 1)
df_agg = df_agg.drop(['Inspection Date'], axis = 1)

In [24]:
df_agg

Restaurant Name  \
0                                   #1 HAWAIIAN BARBEQUE   
1                                                $5 CAFE   
2                    100 DEGREES HOT POT - DELETED 10/15   
3                                     101 MART SNACK BAR   
4                                    168 Market - Bakery   
5                                168 Market - Restaurant   
6                        168 Market / Produce Department   
7                             17 SOUTH BOOZE & BITES BAR   
8                           17 SOUTH BOOZE & BITES GRILL   
9                                     1900 ASIAN CUISINE   
10                                      1923 BOURBON BAR   
11                       1923 BOURBON ISLAND BAR/SUPPORT   
12                                1923 BOURBON STAGE BAR   
13                                  1ST FLOOR BAR- BOMAN   
14                          1ST FLOOR EAST BAR- REYNOLDS   
15                            1ST FLOOR PANTRY- REYNOLDS   
16                          1ST FLOOR WEST BAR- REYNOLDS   
17                                 2 AMIGOS TACOS MOBILE   
18                                            215 Saloon   
19                           24 HOUR MINI MART SNACK BAR   
20                          24 Hour Fitness Smoothie Bar   
21                                             24 LOUNGE   
22                              24X7 MINI MART SNACK BAR   
23                                   28Go the Restaurant   
24                                  2ND FLOOR BAR- BOMAN   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS   
...                                                  ...   
13860                               ZEFFIRINO RESTAURANT   
13861                            ZEFFIRINO SERVICE BAR 1   
13862                         ZENSHIN ASIAN PREP KITCHEN   
13863                           ZENSHIN ASIAN RESTAURANT   
13864                            ZENSHIN ASIAN SUSHI BAR   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE   
13866                            ZEST BISTRO & BAR - BAR   
13867                   ZEST BISTRO & BAR - PREP KITCHEN   
13868                     ZEST BISTRO & BAR - RESTAURANT   
13869                                 ZIKIZ COFFEE HOUSE   
13870                      ZINGERS BEST BURGERS ON EARTH   
13871                  ZINGERS BEST BURGERS ON EARTH BAR   
13872                             ZOMBIE DOGS @ HARD HAT   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...   
13874                                          ZONA ROSA   
13875                                     ZYDECO PO-BOYS   
13876                 Zaytoon Mediterranean - Restaurant   
13877                                          Zizzy Bar   
13878                                   Zizzy Restaurant   
13879                                      Zodies Lounge   
13880              Zoological - Botanical Park Snack Bar   
13881                 at Your Service Catering Snack Bar   
13882                               in N Out Burger #104   
13883                               in N Out Burger #226   
13884                                in N Out Burger #86   
13885                                off the Strip - Bar   
13886                         off the Strip - Restaurant   
13887                        via Brasil Steakhouse - Bar   
13888                   via Brasil Steakhouse - Dishroom   
13889                 via Brasil Steakhouse - Restaurant   

                Permit Number  \
0                 [PR0110047]   
1                 [PR0017837]   
2                 [PR0113718]   
3                 [PR0118514]   
4                 [PR0021447]   
5                 [PR0021446]   
6                 [PR0021445]   
7                 [PR0109450]   
8                 [PR01

In [25]:
dfcategories = df_agg.columns
for x in dfcategories[1:]:
    for y in range(len(df_agg[x])):
        df_agg[x][y] = list(set(df_agg[x][y]))
        df_agg[x][y] = df_agg[x][y][0]

In [26]:
df_agg

Restaurant Name Permit Number  \
0                                   #1 HAWAIIAN BARBEQUE     PR0110047   
1                                                $5 CAFE     PR0017837   
2                    100 DEGREES HOT POT - DELETED 10/15     PR0113718   
3                                     101 MART SNACK BAR     PR0118514   
4                                    168 Market - Bakery     PR0021447   
5                                168 Market - Restaurant     PR0021446   
6                        168 Market / Produce Department     PR0021445   
7                             17 SOUTH BOOZE & BITES BAR     PR0109450   
8                           17 SOUTH BOOZE & BITES GRILL     PR0109449   
9                                     1900 ASIAN CUISINE     PR0099062   
10                                      1923 BOURBON BAR     PR0110263   
11                       1923 BOURBON ISLAND BAR/SUPPORT     PR0108949   
12                                1923 BOURBON STAGE BAR     PR0108950   
13                                  1ST FLOOR BAR- BOMAN     PR0021200   
14                          1ST FLOOR EAST BAR- REYNOLDS     PR0021207   
15                            1ST FLOOR PANTRY- REYNOLDS     PR0021208   
16                          1ST FLOOR WEST BAR- REYNOLDS     PR0021206   
17                                 2 AMIGOS TACOS MOBILE     PR0119469   
18                                            215 Saloon     PR0024098   
19                           24 HOUR MINI MART SNACK BAR     PR0106164   
20                          24 Hour Fitness Smoothie Bar     PR0012767   
21                                             24 LOUNGE     PR0098970   
22                              24X7 MINI MART SNACK BAR     PR0021102   
23                                   28Go the Restaurant     PR0024320   
24                                  2ND FLOOR BAR- BOMAN     PR0021201   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS     PR0021209   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS     PR0021210   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS     PR0104678   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS     PR0021212   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS     PR0021213   
...                                                  ...           ...   
13860                               ZEFFIRINO RESTAURANT     PR0009057   
13861                            ZEFFIRINO SERVICE BAR 1     PR0009059   
13862                         ZENSHIN ASIAN PREP KITCHEN     PR0105401   
13863                           ZENSHIN ASIAN RESTAURANT     PR0098470   
13864                            ZENSHIN ASIAN SUSHI BAR     PR0098471   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE     PR0105403   
13866                            ZEST BISTRO & BAR - BAR     PR0017887   
13867                   ZEST BISTRO & BAR - PREP KITCHEN     PR0098325   
13868                     ZEST BISTRO & BAR - RESTAURANT     PR0017886   
13869                                 ZIKIZ COFFEE HOUSE     PR0120264   
13870                      ZINGERS BEST BURGERS ON EARTH     PR0014610   
13871                  ZINGERS BEST BURGERS ON EARTH BAR     PR0014611   
13872                             ZOMBIE DOGS @ HARD HAT     PR0118834   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...     PR0109033   
13874                                          ZONA ROSA     PR0020519   
13875                                     ZYDECO PO-BOYS     PR0112724   
13876                 Zaytoon Mediterranean - Restaurant     PR0020288   
13877                                          Zizzy Bar     PR0015979   
13878                                   Zizzy Restaurant     PR0015978   
13879                                      Zodies Lounge     PR0007654   
13880              Zoological - Botanical Park Snack Bar     PR0007240   
13881                 at Your Service Catering Snack Bar     PR0005647   
13882                               in N Out 

In [28]:
try:
    with open('df_agg_revs','rb') as f:
        df_agg_revs = pickle.load(f)
        
except FileNotFoundError:
    with open('df_agg_revs','wb') as f:
        pickle.dump(non_0_sick_scores, f)

In [37]:
sick_ids = []
for x in df_agg_revs['business_id'].items():
    sick_ids.append(x[1])
print(sick_ids)

['0tUlPATGNNR7zq5uzh3nog', '9RjV5ZT6Np-VsK96_K1KlQ', '994TtJYxB68eSscnCr94qQ', 'fAJYeE4Q6vwVPGkk1MWCoQ', '2mKC1oXdGCl2w8SAZK8fjw', '9xcOsHNgiLSr18ZmFiU9Qg', 'XqmAbDasWtBd76LpJnMKqQ', 'r164gtGFkTMUqOmZXy2JtQ', 'g2ZRZXEKxj0SL86etbydcg', '3tiST9R0JRLbi4hw4RUBjA', 'wMRx5hvL9_739HYiFLw6Hw', 'vZ2dPk0fPQu4wrXx-DKeDw', 'ETnU0pf3aFuZfpHEHWyd4A', 'gIGZ8QFPlUaxVDTuz3oB9w', 'tc1w6LZUUB9gRMmKYzkIjQ', 'S10V82tt-8EzPF3Y6O7ZbA', 'FTQ1hm8-24_7ou2ILWEyEw', 'JM6CdlbuxfalZZ77DAq7Bg', 'GfuApsBX3jHdZGgWi8PTag', 'bTaIdTtPQIVaj7GChPXGBA', '94HshaJUV3DLvXkvbCbZqg', '0_1am1mjy0hJFyvpsk3x5g', 'LsArM1PhbqvXK3lDK-Ch_w', 'MvnaNl1GEplkpBOBe6nOyw', '2f8o3-FRnq3rwmstUCOZ7A', 'ABzOhfVTcjxwvntaFqHQNw', 'qj2T7GkIVcKCPDFuEqQUrw', 'dRejAgrPIwD6AcykS5Re9g', 'TfW-f8ZtkS5XpIWe9GNZ1g', 'RGEIUdoDtlhlXlkxPKClUg', '24Pyg1ywhjaJDaE9v2tWFQ', 'm66lgElkNdZh4LEDFZ0mNQ', 'PxPJmIAVC7tbEvwQwrmstw', 'LvU4uahiPfogOYL2CmQUDQ', 'Irsdg76tjscSZ2QsvpmYVQ', 'B0xZBB7-oBEiAf_IdqpFWw', '4uDW2Rp9eAPL5BUd5C5-bA', 'TW44bipmQ5fCYbA2D2sajw', 'aGulqXnFmp

In [55]:
file = 'yelp_academic_dataset_business.json'
business_locations = {}
addresses = []
names=[]
with open(file) as f:
    for line in f:
        r_line = json.loads(line)
        if r_line['business_id'] in set(sick_ids):
            business_locations.update({r_line['business_id']:(r_line['full_address']).rsplit('\n', 5)[0].lower()})
            addresses.append(r_line['full_address'].rsplit('\n', 5)[0].lower())
            names.append(r_line['name'].lower())
addresses = np.asarray(addresses)
df_agg_revs['address']=addresses
df_agg_revs['name']=names
#print(business_locations)

In [56]:
df_agg_revs.head()

business_id                                              stars  \
0  0tUlPATGNNR7zq5uzh3nog  [4, 4, 4, 2, 1, 3, 3, 3, 1, 3, 2, 1, 2, 3, 1, ...   
1  9RjV5ZT6Np-VsK96_K1KlQ                                       [3, 1, 5, 4]   
2  994TtJYxB68eSscnCr94qQ  [5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 3, 3, 5, ...   
3  fAJYeE4Q6vwVPGkk1MWCoQ  [2, 3, 1, 3, 4, 5, 5, 2, 5, 2, 4, 4, 4, 4, 3, ...   
4  2mKC1oXdGCl2w8SAZK8fjw  [4, 4, 4, 1, 3, 4, 5, 5, 3, 5, 4, 3, 3, 4, 4, ...   

                                                text  \
0  [With 3 stars as "A-OK" and 4 stars as "Yay!  ...   
1  [Ate at this place a few days ago.  I ordered ...   
2  [This place is fantastic.  The pizza is great,...   
3  [Don't let the newly refurbished decor fool yo...   
4  [Delights Cafe just started serving breakfast ...   

                                               votes  \
0  [{'useful': 2, 'funny': 2, 'cool': 3}, {'usefu...   
1  [{'useful': 0, 'funny': 0, 'cool': 0}, {'usefu...   
2  [{'useful': 3, 'funny': 0, 'cool': 0}, {'usefu...   
3  [{'useful': 0, 'funny': 1, 'cool': 0}, {'usefu...   
4  [{'useful': 0, 'funny': 0, 'cool': 0}, {'usefu...   

                      address                   name  
0     1916 village center cir           chianti cafe  
1       2737 las vegas blvd n             rigos taco  
2             2312 e craig rd  mulligans bar & grill  
3       3100 e lake mead blvd         domino's pizza  
4  1824 e lake mead boulevard  carl's jr restaurants

In [41]:
for x in business_locations:
    print(x)

C3WAtjq006LVC-EHCt-fRA
4O9Du_JiAvz24b2VSOJNTA
jWiGtY7hTqBjbzN3yyRXbw
IV-1NJnGA3O0ZRBrZEWWaA
V4XYRuhq_614tFFvWK5q6g
jgkT9RA3OCBqgniWCkwLrA
-GF267VCCEZ7es8pnKxRoA
1nO6ffHEwZYFBzEsyw1w7w
FV16IeXJp2W6pnghTz2FAw
0BVxe-S5aZMa0GgIYqDmHQ
8zTes4kUSoM161AD7G6DsQ
w29d_3Ua0_JlOaAk89Sr2w
O9ro4MvZ7LDQC76zLEg7AQ
UQEf2T_XYszW2F_y7rBNLA
fuCKyA_ZTXVEo8BwUaeh1Q
tl-yAtKfAB7wtCWehVH1cQ
aGbjLWzcrnEx2ZmMCFm3EA
mb4B1LdcEkz4V6BCm-LeFw
HXmKSaJa7hXsgeCozQs-uQ
XpdKTCAPOSUj50fTnW94aA
OK-v9HfG53SCFvZ4mn5IYQ
1EF9J1TGjxX5gXB3U8hjvA
gM8YcvjG2G8CTgCarkBAeA
uXJbNkU6cKAooMWTWnl5aA
3Igoz7noWksymLqoMWuzRg
shA3pC1JuAkXl28_LVcp7w
O8BRzdXLBzu6K4Sm32fxng
6ZaA_KOS1wW5PqDPAHH9Qg
FDfK4HygC_P-Idoy_K8PKQ
-CIZh4JOZuOEbSzLVrAIGw
YO-hq425916TWm4Rj2-Y-A
N0sUdCBcA3ScYULAkGOqxA
pAWgUsCkPwJG4Vw75Wk0aw
w8sRfLP8U7uX4OktcSbb_w
CfdCsVkzzM3uJKGRzA0ufA
957AgCsV5hhFLDJ5aw546w
cWFpyN49FRsFrNd3_cb50g
XQvKQDAyE3vLsDdjJBKs4Q
X8P2i2tyhLSzljdliBFBrg
sAbRiVqmxNdjTja_vmnsnA
oy90Dd3UjsyAmj_lxSda5g
8buIr1zBCO7OEcAQSZko7w
TNIR0ht13OSqG-HlkoG6uQ
gPmoQaWDiVn

In [58]:
# create lists for addresses of businesses with violations in order to 
# check for overlap between the Las Vegas Violation set and the yelp set.

violation_addresses = []

for line in df_agg['Address']:
    violation_addresses.append({df_agg[str(line).lower())
violation_addresses[:10]

#overlap_df_revs=df_agg_revs[df_agg_revs['address'].isin(violation_addresses)]
#overlap_df_agg=df_agg[df_agg['Address'].isin(overlap_df_revs['address'])]
#overlap_df_revs=overlap_df_revs.reset_index(drop = True)
#overlap_df_revs

business_id  \
0     0aV8mzabKSFylfi0NQGqlQ   
1     qpyD1Bg2JbmQebygwd499g   
2     XGcsblboBZ0Y6VuAfVXxvg   
3     c9ENPUTfi5blUTPgkIHu3w   
4     UZVuNXCNe4yA-gRsayw1VA   
5     unxjJu6JOxVKwGTVdtaXMQ   
6     LZklCs8ZGLOFIeQVcoedgQ   
7     qCvRV9qcHKftXnT9IkmTqA   
8     4kb6cLYeuFqK-HCfCu3IcQ   
9     JQKOBllhtI_X5ZjpWSum6A   
10    cKfp8Wz0321KV5BYEiknlg   
11    e9er1p-8RMyRa9IKUzKE-w   
12    myMbSaVG_C1IKhC5eAWMGg   
13    fe8qs-xZlakS0pqnimL62w   
14    bH6UxhVuIeQ72mBGtPV9dQ   
15    4_OjBv7bu4rRyjo-J1hATw   
16    3r3Nt3ih6RSjtjgcIJXqwQ   
17    DmRw9flaIQ1VTdmjGspN6A   
18    131oUL1FGLOg0AxbBxCtzg   
19    WZHIYPqGPaPQW97GNrkqjQ   
20    nZww1gdBAi9HtRKiQHL2qA   
21    -CIZh4JOZuOEbSzLVrAIGw   
22    jS1niU6w9hSWxdH7wg1iTg   
23    g8MdEDEm61I4TY0cT9O_1A   
24    v-o-ngz-BOQq7aRrwFnX6Q   
25    kAXwEMkpcKMObw45QJGqnw   
26    5LC_KExQFwhEqM1v4Vlceg   
27    A7h8IyuKmNZe-U1O35f0lg   
28    xJwy0A1Z7FcOun8AIvd2ow   
29    YNQgak-ZLtYJQxlDwN-qIg   
...                      ...   
2295  T4AyBSffvi3prLrw0H2enA   
2296  fUQz6bofFVS3pKPJOBQQ5A   
2297  fLQ9XFlHtYHdAKCS4cSMag   
2298  P0Bf-1S_A_g7IbdGuAnG1Q   
2299  0ik871AZ59p77xpkA0RwSA   
2300  3HKp4duvN8lug3ydLHdP6g   
2301  z4NP5n1hb9z2n5uR5YPVsQ   
2302  54HWTbaimVPsPqfUN8-liQ   
2303  f9dxqk9oTmotquKN4vcRCA   
2304  F1Pyhz7l1qNfvfje4gw1Zw   
2305  tOcJnRavMSUzy0EmfyZloA   
2306  kgW2EyuL8d3vA8mkU89A5w   
2307  CEaQ8qPKHrlUzBiZkIPdEg   
2308  Y5C0R1qxUAeIYr1uxOaS5A   
2309  wF2pyzDrzHFED0cXQf-Z6w   
2310  ZeM9AsrVDri_n149dklCXA   
2311  Af2qmVsMCzpKEg4DxsCm2g   
2312  hHneutBDgSpp7l9CUROyaA   
2313  kgJpXOpjH2Fdi0VtdlimiQ   
2314  KuIhhZRZD0mSHJZTD2IU5g   
2315  8q957qOv1wED08Rgs0-VGg   
2316  6OJV--BjqPpJjnTueniqVw   
2317  AAc0nGWLioYSsAyfseMzQg   
2318  hvVckRvOa814ry8q8WGwOg   
2319  Klb2mXlUHuSTyN3CkgZ0EA   
2320  IaUXrUfy2iYrR6dHgvTpUg   
2321  vkOmXXrO-ALfz76TF6s_Gg   
2322  df4A-a0txOfoSBzlBK7q5A   
2323  HekcvFnJCQZjWGrDYD67tw   
2324  uJAhZq0-M9QFdYkaDH8kbQ   

                                                  stars  \
0     [3, 5, 3, 5, 4, 4, 5, 5, 5, 5, 2, 5, 4, 5, 4, ...   
1     [4, 4, 5, 1, 5, 1, 4, 5, 1, 1, 1, 5, 4, 1, 3, ...   
2                                          [4, 5, 5, 5]   
3     [4, 5, 4, 3, 4, 4, 3, 4, 4, 4, 4, 5, 3, 5, 5, ...   
4                                          [3, 4, 3, 2]   
5     [3, 2, 1, 4, 1, 4, 2, 2, 2, 2, 4, 3, 3, 3, 2, ...   
6     [2, 3, 3, 1, 2, 1, 3, 3, 3, 1, 3, 3, 3, 1, 3, ...   
7         [2, 2, 4, 4, 2, 3, 1, 4, 1, 1, 1, 1, 1, 2, 2]   
8                           [1, 1, 3, 5, 1, 1, 1, 1, 1]   
9      [4, 2, 4, 3, 2, 1, 2, 3, 1, 1, 3, 3, 4, 1, 3, 1]   
10                                               [5, 4]   
11    [2, 1, 4, 5, 3, 3, 1, 2, 4, 3, 4, 4, 5, 5, 3, ...   
12    [5, 5, 5, 2, 4, 4, 2, 4, 5, 5, 4, 5, 1, 5, 1, ...   
13    [4, 4, 5, 3, 3, 5, 4, 4, 5, 3, 1, 1, 1, 3, 1, ...   
14    [5, 4, 3, 3, 3, 4, 2, 4, 4, 2, 4, 3, 4, 4, 4, ...   
15                                   [2, 3, 5, 4, 3, 4]   
16    [4, 4, 4, 4, 5, 5, 5, 5, 4, 4, 4, 4, 3, 5, 4, ...   
17    [4, 5, 4, 5, 3, 5, 4, 4, 4, 5, 4, 1, 1, 4, 5, ...   
18    [4, 4, 5, 4, 5, 3, 4, 4, 5, 4, 3, 4, 5, 4, 5, ...   
19    [3, 4, 4, 4, 4, 5, 5, 4, 5, 3, 3, 2, 2, 2, 5, ...   
20    [1, 3, 1, 5, 2, 5, 5, 5, 2, 4, 3, 2, 5, 4, 5, ...   
21           [5, 5, 5, 5, 5, 4, 5, 1, 4, 3, 2, 5, 4, 3]   
22    [5, 3, 5, 5, 5, 4, 5, 5, 5, 5, 2, 4, 5, 5, 3, ...   
23    [1, 3, 1, 1, 2, 2, 5, 2, 2, 1, 3, 3, 5, 5, 5, ...   
24              [5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 5, 5, 5]   
25    [5, 4, 5, 5, 2, 4, 1, 3, 3, 3, 1, 3, 4, 3, 3, ...   
26    [5, 4, 5, 5, 5, 5, 1, 5, 5, 1, 5, 5, 4, 5, 5, ...   
27    [4, 5, 5, 4, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, ...   
28                    [4, 4, 5, 4, 3, 2, 2, 3, 1, 3, 3]   
29    [5, 2, 4, 4, 4, 5, 4, 5, 5, 4, 5, 2, 3, 5, 5, ...   
...                                                 ...   
2295  [5, 4, 5, 5, 4, 4, 5, 1, 4, 4, 5, 4, 5, 5, 5, ...   
2296  [5, 5, 5, 5, 1, 3, 4, 5, 5, 5, 5, 5, 5, 4, 1, ...   
2297                     [

In [59]:
overlap_df_agg

Empty DataFrame
Columns: [Restaurant Name, Permit Number, Location Name, Category Name, Address, Current Demerits, Current Grade, Date Current, Inspection Demerits, Inspection Grade, Permit Status, Inspection Result, Violations, X, Y]
Index: []

In [31]:
def compare()

911


It looks like there are a total of 911 unique restaurant addresses that have an evaluated sick score greater than 0 and entries in the inspection dataset.

In [32]:
categories = df_agg.columns
df_agg = df_agg.drop(['Inspection Grade'], axis=1)
df_agg = df_agg.drop(['Inspection Demerits'], axis=1)              

In [33]:
df_agg

Restaurant Name Permit Number  \
0                                   #1 HAWAIIAN BARBEQUE     PR0110047   
1                                                $5 CAFE     PR0017837   
2                    100 DEGREES HOT POT - DELETED 10/15     PR0113718   
3                                     101 MART SNACK BAR     PR0118514   
4                                    168 Market - Bakery     PR0021447   
5                                168 Market - Restaurant     PR0021446   
6                        168 Market / Produce Department     PR0021445   
7                             17 SOUTH BOOZE & BITES BAR     PR0109450   
8                           17 SOUTH BOOZE & BITES GRILL     PR0109449   
9                                     1900 ASIAN CUISINE     PR0099062   
10                                      1923 BOURBON BAR     PR0110263   
11                       1923 BOURBON ISLAND BAR/SUPPORT     PR0108949   
12                                1923 BOURBON STAGE BAR     PR0108950   
13                                  1ST FLOOR BAR- BOMAN     PR0021200   
14                          1ST FLOOR EAST BAR- REYNOLDS     PR0021207   
15                            1ST FLOOR PANTRY- REYNOLDS     PR0021208   
16                          1ST FLOOR WEST BAR- REYNOLDS     PR0021206   
17                                 2 AMIGOS TACOS MOBILE     PR0119469   
18                                            215 Saloon     PR0024098   
19                           24 HOUR MINI MART SNACK BAR     PR0106164   
20                          24 Hour Fitness Smoothie Bar     PR0012767   
21                                             24 LOUNGE     PR0098970   
22                              24X7 MINI MART SNACK BAR     PR0021102   
23                                   28Go the Restaurant     PR0024320   
24                                  2ND FLOOR BAR- BOMAN     PR0021201   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS     PR0021209   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS     PR0021210   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS     PR0104678   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS     PR0021212   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS     PR0021213   
...                                                  ...           ...   
13860                               ZEFFIRINO RESTAURANT     PR0009057   
13861                            ZEFFIRINO SERVICE BAR 1     PR0009059   
13862                         ZENSHIN ASIAN PREP KITCHEN     PR0105401   
13863                           ZENSHIN ASIAN RESTAURANT     PR0098470   
13864                            ZENSHIN ASIAN SUSHI BAR     PR0098471   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE     PR0105403   
13866                            ZEST BISTRO & BAR - BAR     PR0017887   
13867                   ZEST BISTRO & BAR - PREP KITCHEN     PR0098325   
13868                     ZEST BISTRO & BAR - RESTAURANT     PR0017886   
13869                                 ZIKIZ COFFEE HOUSE     PR0120264   
13870                      ZINGERS BEST BURGERS ON EARTH     PR0014610   
13871                  ZINGERS BEST BURGERS ON EARTH BAR     PR0014611   
13872                             ZOMBIE DOGS @ HARD HAT     PR0118834   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...     PR0109033   
13874                                          ZONA ROSA     PR0020519   
13875                                     ZYDECO PO-BOYS     PR0112724   
13876                 Zaytoon Mediterranean - Restaurant     PR0020288   
13877                                          Zizzy Bar     PR0015979   
13878                                   Zizzy Restaurant     PR0015978   
13879                                      Zodies Lounge     PR0007654   
13880              Zoological - Botanical Park Snack Bar     PR0007240   
13881                 at Your Service Catering Snack Bar     PR0005647   
13882                               in N Out 

In [73]:
df_cat = pd.get_dummies(df_agg['Category Name'], prefix = 'category')
df_grade = pd.get_dummies(df_agg['Current Grade'], prefix = 'grade_cur')
df_perm = pd.get_dummies(df_agg['Permit Status'], prefix = 'permit')
df_res = pd.get_dummies(df_agg['Inspection Result'], prefix = 'result')
df_dems = pd.get_dummies(df_agg['Current Demerits'], prefix = 'demerits')
df_type = pd.get_dummies(df_agg['Inspection Type'], prefix = 'type')
df_new = pd.concat([df_agg, df_cat], axis=1)
df_new = pd.concat([df_new, df_grade], axis=1)
df_new = pd.concat([df_new, df_perm], axis=1)
df_new = pd.concat([df_new, df_res], axis=1)
df_new = pd.concat([df_new, df_dems], axis=1)
df_new = pd.concat([df_new, df_type], axis=1)
df_new = df_new.drop(['Category Name'], axis = 1 )
df_new = df_new.drop(['Current Grade'], axis = 1 )
df_new = df_new.drop(['Permit Status'], axis = 1 )
df_new = df_new.drop(['Inspection Result'], axis = 1)
df_new = df_new.drop(['Current Demerits'], axis = 1)
df_new = df_new.drop(['Inspection Type'], axis = 1)
df_new = df_new.drop(['Date Current'], axis = 1)
df_new = df_new.drop(['Restaurant Name'], axis = 1)
df_new = df_new.drop(['Permit Number'], axis = 1)
df_new = df_new.drop(['Location Name'], axis = 1)
df_new = df_new.drop(['Address'], axis = 1)
df_new.fillna(0)

KeyError: 'Inspection Type'

In [ ]:
df_new

In [60]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.cross_validation import train_test_split

X = df_new.loc[:,:].values
y = df_new['grade_cur_X'].values
X_train, X_test, y_train, y_test = train_test_split(X, y, )
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))


[ 0.  0.  0. ...,  0.  0.  0.]
Accuracy on test data: 0.9997120644975526


In [62]:
X = df_new.loc[:,:].values
y = df_new['grade_cur_A'].values
X_train, X_test, y_train, y_test = train_test_split(X, y)
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))

[ 1.  1.  1. ...,  1.  1.  1.]
Accuracy on test data: 0.9979844514828679
